### DISCUSSION TITLE
How can i insert queries in django using views.py ?

### DISCUSSION THREAD

**Discussion by Ekata2003** [Ekata2003/Social-media-Django#13] [Q&A] [answered]
How can i insert queries in django using views.py ?

**Comment by prasad-chavan1** [comment, marked as answer]
# Inserting Queries in Django Using views.py

To insert data into the database in Django, you typically use views.py to handle user input and create database records. Here's a step-by-step guide on how to do this:

## Step 1: Import Necessary Modules

In your `views.py`, start by importing the necessary modules and models:

```python
from django.shortcuts import render, redirect
from .models import YourModel

```
  def create_record(request):
    if request.method == 'POST':
        # Get data from the POST request
        data_to_insert = request.POST.get('data_field_name')
        
        # Create a new record in the database
        new_record = YourModel(data_field=data_to_insert)
        new_record.save()

        # Redirect to a success page or another URL
        return redirect('success_page')
    
    return render(request, 'your_template.html')
